### PR TITLE
Belos:  Add more Tpetra tests for Block CG/GMRES, update Tpetra syntax

### PR DESCRIPTION
**Pull request by hkthorn**
@trilinos/belos 

## Description

Upon reviewing the enabled tests, it was determined that the block GMRES tests were completely disabled for the Tpetra interface and some useful block CG tests were also disabled for the Tpetra interface. The single test that existed in the tpetra/test/BlockGmres directory used a very large linear system and did not run correctly when it was enabled. That test has been updated to use a smaller test matrix from the Harwell-Boeing library and a second test has been added to extend the solver coverage for the Tpetra interface. There was a complex-valued block CG test that was disabled for the Tpetra interface, which has not been enabled.

Additionally, this pull request updates Tpetra syntax to current best practices for all the Tpetra tests.

## Motivation and Context

The block GMRES solver was not being tested for the Tpetra linear algebra interface and the block CG had limited testing. This enables better code coverage for an important linear algebra interface.

## How Has This Been Tested?

Serial & Parallel OSX with GCC 7.x.

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/belos/tpetra/test/BlockCG/CMakeLists.txt [hunk 1]
```diff
@@ -21,37 +21,24 @@ TRIBITS_ADD_EXECUTABLE_AND_TEST(
...
-#  EXEDEPS Tpetra_BlockCG_complex_hb_test
-#  )
-
-TRIBITS_COPY_FILES_TO_BINARY_DIR(Tpetra_BlockCG_hb_CopyFiles1
-  SOURCE_DIR ${Belos_SOURCE_DIR}/tpetra/test/BlockCG
-  SOURCE_FILES bcsstk17.rsa 
-  EXEDEPS Tpetra_BlockCG_hb_test
-  # EXEDEPS Tpetra_BlockCG_hb_test Tpetra_BlockCG_hb_multiprec_test
-  )
+# Tpetra matrix file reader does not read complex-valued matrices 6/28/2018
```
Comment (mhoemmen) [inline packages/belos/tpetra/test/BlockCG/CMakeLists.txt]: The Matrix Market format reader can read complex-valued matrices just fine.  Is there a Matrix Market version of this matrix?

**Comment by hkthorn** [comment]
@mhoemmen I found a small issue in the CMakeLists.txt that should conditionally compile the double-float test.  This was causing the build error.

File: packages/belos/tpetra/test/BlockGmres/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,32 @@
 
 
+ASSERT_DEFINED(Anasazi_SOURCE_DIR)
 ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_Triutils)
+ASSERT_DEFINED(Tpetra_INST_FLOAT)
+
 IF (${PACKAGE_NAME}_ENABLE_Triutils)
+  IF (Tpetra_INST_FLOAT)
```
Comment (mhoemmen) [inline packages/belos/tpetra/test/BlockGmres/CMakeLists.txt]: I don't think any of the Dashboard builds test `float`, alas, so you may have to enable it locally to get test coverage.

File: packages/belos/tpetra/test/BlockCG/CMakeLists.txt (hunk 1 above)
Comment (hkthorn) [reply packages/belos/tpetra/test/BlockCG/CMakeLists.txt]: Teuchos::GlobalMPISession::GlobalMPISession(): started serial run
terminate called after throwing an instance of 'std::runtime_error'
  what():  /Users/hkthorn/Trilinos/TrilinosFork/Trilinos/packages/tpetra/core/inout/Tpetra_MatrixIO.cpp:318:

Throw number = 2

Throw test that evaluated to true: true

Tpetra::Utils::readHBInfo() of filename "mhd1280b.mtx" caught exception:
/Users/hkthorn/Trilinos/TrilinosFork/Trilinos/packages/tpetra/core/inout/Tpetra_MatrixIO.cpp:127:

Throw number = 1

Throw test that evaluated to true: std::sscanf(line,"%14d%14d%14d%14d",&Totcrd,&Ptrcrd,&Indcrd,&Valcrd) != 4

Tpetra::Utils::readHBHeader(): Improperly formatted H/B file: error reading pointers (line 2)

**Pull request merged**